src/sqlparser/tests/sqlparser_common.rs

```diff
@@ -3038,13 +3046,66 @@ fn parse_create_materialized_view() {
             query,
             materialized,
             with_options,
+            emit_mode,
+        } => {
+            assert_eq!("myschema.myview", name.to_string());
+            assert_eq!(Vec::<Ident>::new(), columns);
+            assert_eq!("SELECT foo FROM bar", query.to_string());
+            assert!(materialized);
+            assert_eq!(with_options, vec![]);
+            assert!(!or_replace);
+            assert_eq!(emit_mode, None);
+        }
+        _ => unreachable!(),
+    }
+}
+
+#[test]
+fn parse_create_materialized_view_emit_immediately() {
+    let sql = "CREATE MATERIALIZED VIEW myschema.myview EMIT IMMEDIATELY AS SELECT foo FROM bar";
```

Comment:
I thought it should be part of the query i.e. inside the `SELECT ... ` clause. Did we change our mind?

Reply:
If we defined them on `SELECT`, how to define the semantics on subquery or CTE?

Currently, our conclusion is that we should ensure a mview or a sink can only have one emit option, the only way is defined on view.

Reply:
https://github.com/risingwavelabs/rfcs/pull/30 defines the semantics in terms of relational algebra (i.e. given input set, what should be the output set), so the semantics is still well-defined for subqueries or CTEs.